### PR TITLE
Remove guaranteed compatibility with versions to 10> and <15 

### PR DESCRIPTION
**Pull request by bidoubiwa**
fixes: #346

Since the builded package can be used (without guarantee) in other node versions, we should not make it mandatory to use the right node version. It is at the discretion of the user. 

It is how most libraries handle node support. The use of `engines` in packages json limits the possibility to install the package in a specific node environment (see [issue](https://github.com/meilisearch/documentation/issues/963#issuecomment-854264913)).
Whereas typically a package that is bundled and transpiled should be as compatible as possible with all js environments. But, the packages we use in development are not. This is why we add it in the CONTRIBUTING guide

### CODE REVIEW

**Review by mdubus — DISMISSED** [review]

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -11,25 +11,23 @@ on:
 
 jobs:
   tests:
-    name: tests
     runs-on: ubuntu-latest
+    strategy:
```
Comment (curquiza) [inline .github/workflows/tests.yml:15]:
```suggestion
    strategy:
      fail-fast: false
```

Comment (curquiza) [reply .github/workflows/tests.yml:15]: This way, if one of the tests fails (for example node 12), the other tests still run.

File: bors.toml [hunk 2]
```diff
@@ -1,3 +1,8 @@
-status = ['tests', 'linter-check']
+status = [
+    'tests',
```
Comment (curquiza) [inline bors.toml:2]:
```suggestion
```

**Review by curquiza — CHANGES_REQUESTED** [review]

**Comment by curquiza** [comment]
⚠️ This PR will deserve a new minor in the versionning.

**Review by curquiza — APPROVED** [review]

**Comment by bidoubiwa** [comment]
bors merge

**Pull request merged**